### PR TITLE
IOS-XR: Add iosxr_static_routes resource module

### PR DESCRIPTION
**Pull request by NilashishC**
##### SUMMARY
- Add iosxr_static_routes resource module
- Targets https://github.com/ansible-network/resource_module_models/pull/59

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
iosxr_static_routes.py

### CODE REVIEW

File: lib/ansible/module_utils/network/iosxr/config/static_routes/static_routes.py [hunk 1]
```diff
@@ -0,0 +1,599 @@
...
+
+        if self.state in self.ACTION_STATES or self.state == "gathered":
+            changed_static_routes_facts = self.get_static_routes_facts()
+
+        elif self.state == "rendered":
+            result["rendered"] = commands
+
+        elif self.state == "parsed":
+            running_config = self._module.params["running_config"]
+            if self.state == "parsed" and not running_config:
```
Comment (rohitthakur2590) [inline lib/ansible/module_utils/network/iosxr/config/static_routes/static_routes.py]: redundant state check

File: lib/ansible/modules/network/iosxr/iosxr_static_routes.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+def main():
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
+    required_if = [
+        ("state", "merged", ("config",)),
+        ("state", "replaced", ("config",)),
+        ("state", "overridden", ("config",)),
```
Comment (rohitthakur2590) [inline lib/ansible/modules/network/iosxr/iosxr_static_routes.py:949]: can we have entry for parsed state that requires ('running_config')

File: lib/ansible/module_utils/network/iosxr/config/static_routes/static_routes.py (hunk 1 above)
Comment (NilashishC) [reply lib/ansible/module_utils/network/iosxr/config/static_routes/static_routes.py]: Resolved.

File: lib/ansible/modules/network/iosxr/iosxr_static_routes.py (hunk 2 above)
Comment (NilashishC) [reply lib/ansible/modules/network/iosxr/iosxr_static_routes.py:949]: Resolved.

**Comment by NilashishC** [comment]
The one failure in Py3.7 test is related to `Socket is closed` issue. `iosxr_static_routes` is green.

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
Again a `Socket is closed` issue with legacy `iosxr_interface`. But rest all green.

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
New failures are again related to CI infrastructure.

**Comment by NilashishC** [comment]
@pabelanger suggestions?^^

**Comment by pabelanger** [comment]
TIMED_OUT is likely related to the job running on a slow hypervisor. We need somebody to help debug which node is slow and talk with our nodepool provider.

File: test/integration/targets/iosxr_static_routes/tests/cli/empty_config.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+- name: Parsed with empty running_config should give appropriate error message
+  iosxr_static_routes:
+    running_config:
+    state: parsed
+  register: result
+  ignore_errors: True
+
+- assert:
+    that:
+      - result.msg == 'value of running_config parameter must not be empty for state parsed'
```
Comment (rohitthakur2590) [inline test/integration/targets/iosxr_static_routes/tests/cli/empty_config.yaml:47]: we can have rendered as well right ?

File: lib/ansible/module_utils/network/iosxr/config/static_routes/static_routes.py [hunk 4]
```diff
@@ -0,0 +1,599 @@
...
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        if self.state in ("overridden", "merged", "replaced") and not want:
```
Comment (rohitthakur2590) [inline lib/ansible/module_utils/network/iosxr/config/static_routes/static_routes.py]: 'rendered' needs to be included.

**Comment by NilashishC** [comment]
iosxr_static_routes runs are green, CI failing with unrelated issues.

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
> Build succeeded (third-party-check pipeline).
> 
> * [ansible-test-network-integration-iosxr-python27 ](https://dashboard.zuul.ansible.com/t/ansible/build/66a6ae2139904f06852b2955952a81f2) : FAILURE in 2h 02m 38s (non-voting)
> * [ansible-test-network-integration-iosxr-python35 ](https://dashboard.zuul.ansible.com/t/ansible/build/c308e93f09dd47f5ad6f00f23616af81) : FAILURE in 2h 30m 15s (non-voting)
> * [ansible-test-network-integration-iosxr-python36 ](https://dashboard.zuul.ansible.com/t/ansible/build/0257dc0692cc43169ebabe3b5bd79ebb) : FAILURE in 2h 08m 23s (non-voting)
> * [ansible-test-network-integration-iosxr-python37 ](https://dashboard.zuul.ansible.com/t/ansible/build/d270da11b0d9400aa38839ca1ae4d260) : TIMED_OUT in 2h 33m 27s (non-voting)
> * [ansible-test-network-integration-iosxr-python38 ](https://dashboard.zuul.ansible.com/t/ansible/build/ea793d2c464642029e4d0dba7315fec2) : FAILURE in 2h 09m 05s (non-voting)

Made a mistake while writing unit tests. Fixed now.

**Review by rohitthakur2590 — APPROVED** [review]

**Comment by NilashishC** [comment]
Unrelated tests failing with `Socket is closed`. All iosxr_static_routes tests are green.

**Pull request merged**